dags/analyzer_helper/telegram/utils/date_time_format_converter.py

```
@@ -0,0 +1,38 @@
+    def string_to_datetime(date_string: str) -> datetime:
+        """
+        Convert an ISO format date string to a Python datetime object.
+        
+        :param date_string: The date string to convert (e.g., "2023-06-01T12:00:00").
+        :return: The corresponding datetime object.
+        :raises ValueError: If the input string is not in a valid ISO format.
+        """
+        try:
+            return datetime.fromisoformat(date_string)
+        except ValueError as e:
+            raise ValueError(f"Invalid date string format: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)